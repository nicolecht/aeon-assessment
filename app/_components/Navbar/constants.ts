export const navLinks = [
  { name: "Showcase", href: "/showcase" },
  { name: "Docs", href: "/docs" },
  { name: "Blog", href: "/blog" },
  { name: "Analytics", href: "/analytics" },
  { name: "Commerce", href: "/commerce" },
  { name: "Templates", href: "/templates" },
  { name: "Enterprise", href: "/enterprise" },
];
