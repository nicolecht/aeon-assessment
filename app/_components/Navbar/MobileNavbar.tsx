"use client";

import { useState } from "react";
import Link from "next/link";
import Image from "next/image";
import { navLinks } from "./constants";
import LinkButton from "../LinkButton";
import Input from "../Input";

export default function MobileNavbar() {
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const [isSearchBarOpen, setIsSearchBarOpen] = useState(false);

  const toggleMenu = () => {
    setIsMenuOpen((prev) => !prev);
    setIsSearchBarOpen(false);
  };

  const toggleSearchBar = () => {
    setIsSearchBarOpen((prev) => !prev);
    setIsMenuOpen(false);
  };

  const closeMenuAndSearchBar = () => {
    setIsMenuOpen(false);
    setIsSearchBarOpen(false);
  };

  return (
    <nav className="fixed w-full lg:hidden">
      {/* Navbar Header */}
      <div className="flex justify-between p-4">
        <Link
          href="/"
          className="text-primary-500 font-semibold"
          onClick={closeMenuAndSearchBar}
        >
          Aeon Bank Assessment
        </Link>

        {/* Search & Menu Buttons */}
        <div className="flex items-center space-x-4">
          <button onClick={toggleSearchBar}>
            <Image
              src="/icons/magnifying-glass-solid.svg"
              alt="Toggle search"
              width={16}
              height={16}
            />
          </button>

          <button onClick={toggleMenu}>
            <Image
              src={isMenuOpen ? "/icons/xmark-solid.svg" : "/icons/bars-solid.svg"}
              alt="Toggle navigation menu"
              width={16}
              height={16}
            />
          </button>
        </div>
      </div>

      {/* Mobile Menu */}
      <div
        className={`transition-all duration-500 bg-primary-100 px-4 overflow-hidden ${
          isMenuOpen ? "max-h-screen opacity-100" : "max-h-0 opacity-0"
        }`}
      >
        <div className="flex flex-col justify-between py-4 space-y-4">
          {navLinks.map((link) => (
            <Link
              key={link.href}
              href={link.href}
              className="font-semibold text-secondary-500 hover:text-primary-500"
              onClick={() => setIsMenuOpen(false)}
            >
              {link.name}
            </Link>
          ))}
          <LinkButton href="/login" onClick={closeMenuAndSearchBar}>
            Login
          </LinkButton>
        </div>
      </div>

      {/* Mobile Search Bar */}
      <div
        className={`transition-all duration-300 px-4 overflow-hidden ${
          isSearchBarOpen ? "max-h-screen opacity-100" : "max-h-0 opacity-0"
        }`}
      >
        <Input
          type="search"
          placeholder="Search..."
          className="border-x-0 border-t-0 rounded-none"
        />
      </div>
    </nav>
  );
}
