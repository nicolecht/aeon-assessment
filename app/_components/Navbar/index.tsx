import Link from "next/link";
import MobileNavbar from "./MobileNavbar";
import { navLinks } from "./constants";
import LinkButton from "../LinkButton";
import Input from "../Input";

export default function Navbar() {
  return (
    <>
      <nav
        data-testid="desktop-navbar"
        className="fixed hidden w-full p-4 lg:flex lg:justify-between"
      >
        <div className="flex items-center space-x-4 font-semibold grow">
          <Link href="/" className="text-primary-500">
            Aeon Bank Assessment
          </Link>
          {navLinks.map((link) => (
            <Link
              key={link.name}
              href={link.href}
              className="text-secondary-500 hover:text-primary-500"
            >
              {link.name}
            </Link>
          ))}
        </div>

        <div className="flex items-center space-x-4">
          <Input type="search" placeholder="Search..." />
          <LinkButton href="/login">Login</LinkButton>
        </div>
      </nav>

      <MobileNavbar />
    </>
  );
}
