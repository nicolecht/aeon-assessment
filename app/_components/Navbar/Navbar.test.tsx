import { render, screen } from "@testing-library/react";
import Navbar from "../Navbar";
import { navLinks } from "./constants";

describe("Navbar component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders desktop navbar with correct elements", () => {
    render(<Navbar />);

    expect(screen.getByTestId("desktop-navbar")).toBeInTheDocument();

    navLinks.forEach(({ name }) => {
      expect(screen.getAllByText(name)[0]).toBeInTheDocument();
    });

    expect(screen.getAllByPlaceholderText("Search...")[0]).toBeInTheDocument();
    expect(screen.getAllByText("Login")[0]).toBeInTheDocument();
  });
});
