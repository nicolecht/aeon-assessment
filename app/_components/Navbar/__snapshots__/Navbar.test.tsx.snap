// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="fixed hidden w-full p-4 lg:flex lg:justify-between"
    data-testid="desktop-navbar"
  >
    <div
      class="flex items-center space-x-4 font-semibold grow"
    >
      <a
        class="text-primary-500"
        href="/"
      >
        Aeon Bank Assessment
      </a>
      <a
        class="text-secondary-500 hover:text-primary-500"
        href="/showcase"
      >
        Showcase
      </a>
      <a
        class="text-secondary-500 hover:text-primary-500"
        href="/docs"
      >
        Docs
      </a>
      <a
        class="text-secondary-500 hover:text-primary-500"
        href="/blog"
      >
        Blog
      </a>
      <a
        class="text-secondary-500 hover:text-primary-500"
        href="/analytics"
      >
        Analytics
      </a>
      <a
        class="text-secondary-500 hover:text-primary-500"
        href="/commerce"
      >
        Commerce
      </a>
      <a
        class="text-secondary-500 hover:text-primary-500"
        href="/templates"
      >
        Templates
      </a>
      <a
        class="text-secondary-500 hover:text-primary-500"
        href="/enterprise"
      >
        Enterprise
      </a>
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <input
        class="p-2 border rounded-lg w-full outline-none border-neutral-300 focus:border-neutral-500 undefined"
        placeholder="Search..."
        type="search"
      />
      <a
        class="p-3 font-semibold text-center text-white rounded-full min-w-32 bg-primary-500 hover:bg-primary-600 undefined"
        href="/login"
      >
        Login
      </a>
    </div>
  </nav>
  <nav
    class="fixed w-full lg:hidden"
    data-testid="mobile-navbar"
  >
    <div
      class="flex justify-between p-4"
    >
      <a
        class="font-semibold text-primary-500"
        href="/"
      >
        Aeon Bank Assessment
      </a>
      <div
        class="flex items-center space-x-4"
      >
        <button>
          <img
            alt="Toggle search"
            data-nimg="1"
            decoding="async"
            height="16"
            loading="lazy"
            src="/icons/magnifying-glass-solid.svg"
            style="color: transparent;"
            width="16"
          />
        </button>
        <button>
          <img
            alt="Toggle navigation menu"
            data-nimg="1"
            decoding="async"
            height="16"
            loading="lazy"
            src="/icons/bars-solid.svg"
            style="color: transparent;"
            width="16"
          />
        </button>
      </div>
    </div>
    <div
      class="transition-all duration-500 bg-primary-100 px-4 overflow-hidden max-h-0 opacity-0"
      data-testid="mobile-menu"
    >
      <div
        class="flex flex-col justify-between py-4 space-y-4"
      >
        <a
          class="font-semibold text-secondary-500 hover:text-primary-500"
          href="/showcase"
        >
          Showcase
        </a>
        <a
          class="font-semibold text-secondary-500 hover:text-primary-500"
          href="/docs"
        >
          Docs
        </a>
        <a
          class="font-semibold text-secondary-500 hover:text-primary-500"
          href="/blog"
        >
          Blog
        </a>
        <a
          class="font-semibold text-secondary-500 hover:text-primary-500"
          href="/analytics"
        >
          Analytics
        </a>
        <a
          class="font-semibold text-secondary-500 hover:text-primary-500"
          href="/commerce"
        >
          Commerce
        </a>
        <a
          class="font-semibold text-secondary-500 hover:text-primary-500"
          href="/templates"
        >
          Templates
        </a>
        <a
          class="font-semibold text-secondary-500 hover:text-primary-500"
          href="/enterprise"
        >
          Enterprise
        </a>
        <a
          class="p-3 font-semibold text-center text-white rounded-full min-w-32 bg-primary-500 hover:bg-primary-600 undefined"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
    <div
      class="transition-all duration-300 px-4 overflow-hidden max-h-0 opacity-0"
    >
      <input
        class="p-2 border rounded-lg w-full outline-none border-neutral-300 focus:border-neutral-500 border-t-0 rounded-none border-x-0"
        placeholder="Search..."
        type="search"
      />
    </div>
  </nav>
</DocumentFragment>
`;
